atom/common/api/atom_api_shell.cc

```diff
@@ -30,6 +58,61 @@ bool OpenExternal(
   return platform_util::OpenExternal(url, activate);
 }
 
+#if defined(OS_WIN)
+bool WriteShortcutLink(const base::FilePath& shortcut_path,
+                       mate::Arguments* args) {
+  base::win::ShortcutOperation operation = base::win::SHORTCUT_CREATE_ALWAYS;
+  args->GetNext(&operation);
+  mate::Dictionary options = mate::Dictionary::CreateEmpty(args->isolate());
+  if (!args->GetNext(&options)) {
+    args->ThrowError();
+    return false;
+  }
+
+  base::win::ShortcutProperties properties;
+  base::FilePath path;
+  base::string16 str;
+  int index;
+  if (options.Get("target", &path))
+    properties.set_target(path);
+  if (options.Get("cwd", &path))
+    properties.set_working_dir(path);
+  if (options.Get("args", &str))
+    properties.set_arguments(str);
+  if (options.Get("description", &str))
+    properties.set_description(str);
+  if (options.Get("icon", &path) && options.Get("iconIndex", &index))
```

Comment:
@zcbenz The docs indicate that `iconIndex` has a default of 0 but this appears to not set the icon if `iconIndex` is not set.

Which behavior is expected / wanted and I will fix whichever end is wrong 😄